### PR TITLE
[Repo] Add repo test case to evaluate dynamisim - @open sesame 12/04 11:10

### PR DESCRIPTION
**Pull request by jijoongmoon**
# PR Description

In order to change the slot-id dynamically, nnstreamer_test_dynamicity
is added. It has simple gstreamer application to change slot id.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: tests/nnstreamer_repo_dynamicity/tensor_repo_dynamic_test.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/**
+ * @file    tensor_repo_dynamic_test.c
+ * @date    03 Dec 2018
+ * @brief   test case to test tensor repo plugin dynamism
+ * @see     https://github.com/nnsuite/nnstreamer
+ * @author  Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug     No known bugs except NYI.
+ */
```
Comment (myungjoo) [inline tests/nnstreamer_repo_dynamicity/tensor_repo_dynamic_test.c]: Please draw the pipeline schematics as a comment in the code.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -36,6 +36,11 @@ ADD_EXECUTABLE(unittest_sink nnstreamer_sink/unittest_sink.cpp ${gtestSrc})
 TARGET_INCLUDE_DIRECTORIES(unittest_sink PUBLIC ${pkgs_INCLUDE_DIRS} ${testpkgs_INCLUDE_DIRS} ${gtestInc})
 TARGET_LINK_LIBRARIES(unittest_sink nnstreamer ${pkgs_LIBRARIES} ${testpkgs_LIBRARIES} ${gtestLink})
 
+# Unit test using tensor_repo_dynamic
+ADD_EXECUTABLE(test_repo_dynamic nnstreamer_repo_dynamicity/tensor_repo_dynamic_test.c)
+TARGET_INCLUDE_DIRECTORIES(test_repo_dynamic PUBLIC ${pkgs_INCLUDE_DIRS} ${testpkgs_INCLUDE_DIRS})
+TARGET_LINK_LIBRARIES(test_repo_dynamic nnstreamer ${pkgs_LIBRARIES} ${testpkgs_LIBRARIES})
+
```
Comment (myungjoo) [inline tests/CMakeLists.txt]: Please update the .spec file so that it runs test_repo_dynamic as a unit test.

**Comment by myungjoo** [comment]
Besides, please update the meson build script as well from now on. (From Sprint 1812A)

**Comment by jijoongmoon** [comment]
For now, this is just simple pipeline without testing tensor_repo dynamism. In order to test, the function which changes the property "slot-index" at runtime and some more implementation in tensor_repo.c and some codes. Next PR is already ready including what it should be done and I will update what mentioned here. This PR and some fixes will be included as one of the commit. So, I close this PR.

**Pull request closed without merging**